### PR TITLE
Issues/35602 AlertProcessor - Argument #2 ($storeId) must be of type int, string given

### LINKED ISSUE #35602: AlertProcessor - Argument #2 ($storeId) must be of type int, string given
### Preconditions and environment

- Magento version 2.4.4
- PHP8.1

### Steps to reproduce
- Enable product alert stock in BO (Configuration > Catalog >Product Alerts > Allow Alert When Product Comes Back in Stock > Yes, Allow Alert When Product Price Changes - Yes)
- Reindex and cache flush
- Front end - Login as customer - Select a Product without stock - Register to a product stock alert
(Notify me when the price drops, Notify me when this product is in stock)
- Add stock to that product from admin
- Launch consumers

### Expected result

Alert stock email being sent

### Actual result
Emails not triggered for Product Alerts

```
[2022-06-10T07:31:56.520650+00:00] .CRITICAL: Magento\ProductAlert\Model\Mailing\ErrorEmailSender::execute(): Argument #2 ($storeId) must be of type int, string given, called in /var/www/html/vendor/magento/module-product-alert/Model/Mailing/AlertProcessor.php on line 130 [] []
```

### Additional information

I guess (not tested) that change could solve the issue 
```
Magento\Store\Model\Store

ACTUAL CODE
   /**
     * Get store identifier
     *
     * @return  int
     */
    public function getId()
    {
        return $this->_getData(self::STORE_ID);
    }


PROPOSED FIX

   /**
     * Get store identifier
     *
     * @return  int
     */
    public function getId(): int
    {
        return (int) $this->_getData(self::STORE_ID);
    }

```
(of course, StoreInterface interface should be modified accordingly.)

### Release note

_No response_

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [X] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by artskirk**
### Description (*)

Emails not triggered for Product Alerts. `CRITICAL: Magento\ProductAlert\Model\Mailing\ErrorEmailSender::execute(): Argument #2 ($storeId) must be of type int, string given, called in /var/www/html/vendor/magento/module-product-alert/Model/Mailing/AlertProcessor.php on line 130`

I have updated return type statement from string to integer in app/code/Magento/Store/Model/Store.php::getId() as per doc.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#35602

### Manual testing scenarios (*)

- Enable product alert stock in BO (Configuration > Catalog >Product Alerts > Allow Alert When Product Comes Back in Stock > Yes, Allow Alert When Product Price Changes - Yes)
- Reindex and cache flush
- Front end - Login as customer - Select a Product without stock - Register to a product stock alert
(Notify me when the price drops, Notify me when this product is in stock)
- Add stock to that product from admin
- Launch consumers

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by artskirk** [comment]
@magento run all tests

File: app/code/Magento/Store/Test/Unit/Model/StoreTest.php [hunk 1]
```diff
@@ -162,6 +162,25 @@ public function testSetWebsite()
...
+    {
+        $testId = 1;
+        $websiteRepository = $this->getMockBuilder(WebsiteRepositoryInterface::class)
+            ->getMockForAbstractClass();
+
+        /** @var Store $model */
+        $model = $this->objectManagerHelper->getObject(
+            Store::class,
+            ['websiteRepository' => $websiteRepository]
+        );
```
Comment (Den4ik) [inline app/code/Magento/Store/Test/Unit/Model/StoreTest.php]: Please avoid to use ObjectManager helper in the new tests, because it's deprecated.

**Review by Den4ik — CHANGES_REQUESTED** [review]
@artskirk Thanks for your PR.
Please look at my comment below

File: app/code/Magento/Store/Test/Unit/Model/StoreTest.php (hunk 1 above)
Comment (artskirk) [reply app/code/Magento/Store/Test/Unit/Model/StoreTest.php]: Thanks for your feedback. Fixed.

**Comment by Den4ik** [comment]
@magento run all tests

File: app/code/Magento/Store/Model/Store.php [hunk 2]
```diff
@@ -766,7 +766,7 @@ public function isUseStoreInUrl()
      *
      * @return  int
      */
-    public function getId()
+    public function getId() :int
```
Comment (Den4ik) [inline app/code/Magento/Store/Model/Store.php:769]: There is backward incompatible changes due to due to `@api` mark for class.
In additional return type declaration don't fix problem and type should be converted in the body of the function `return (int)$this->_getData(self::STORE_ID);`

**Review by Den4ik — CHANGES_REQUESTED** [review]
Hi @artskirk 
Thanks for you contribution. Please look at  my comment

**Comment by ishakhsuvarov** [comment]
Hi

I am closing this PR for now as it seems like it has become stale and still needs work.

Please feel free to reopen if you wish to continue.

Thanks!

**Pull request closed without merging**